### ISSUE TITLE
Need help enter token of Funcaptcha

### ISSUE THREAD

**Issue by MacMarde** [ad-m/python-anticaptcha#86]
I try to create an account with Microsoft. https://login.live.com/

Sometimes I do get a FunCaptcha to solve. I am able to solve the captcha und get a valid token.

I try to enter the token like this from the examples:

```
script = """
    document.getElementById('FunCaptcha-Token').value = decodeURIComponent('{0}');
    document.getElementById('verification-token').value = decodeURIComponent('{0}');    
    """.format(
        quote(token)
    )
    browser.execute_script(script)
```

I can see with chrome developer-tools that the elements "FunCaptcha-Token" and "verification-token" are present on the website.

Also the token I do receive from anticaptcha is not empty.

Unfortunately I do get this exception message:
selenium.common.exceptions.JavascriptException: Message: javascript error: Cannot set property 'value' of null

The elments "FunCaptcha-Token" and "verification-token" are inside of a iframe with id "enforcementFrame" and there is also a "#document" shadow DOM. In selenium this needs a workaround, but I do not know if the above Javascript function can handle this. (See the screenshot)

I really need help to enter the tokens und also to submit the form. It seems to be not trivial. 

![image](https://user-images.githubusercontent.com/50266741/96356251-32784080-10ec-11eb-90bd-b9b9c10ca7cb.png)

**Comment by MacMarde** [comment]
I have found a partial solution:

I did use:

`browser.switch_to_frame(browser.find_element_by_id('enforcementFrame'))`

Afer I have switched to this iframe "enforcementFrame" I can now use the above javascript function to enter the valid tokens and it is working fine.

But I still have no idea how to submit. The button seems to be located on a canvas element and I believe therefore I can not find this button as an element.
I guess it is possible to use selenium ActionChains to click the initial "Next" button, but the follwing "Done" button can only be clicked after a click on the arrows to turn the image. 

I am not sure if this is possible with ActionChains. Probably there is a better solution possible?

**Comment by MacMarde** [comment]
I think I have found the relevant website code. But I do not know what to do with it. Any help welcome.

![funcaptcha_callback](https://user-images.githubusercontent.com/50266741/97261350-f006e000-181e-11eb-8dcc-b2f406b63dc2.png)

**Comment by scullcandyz** [comment]
> I think I have found the relevant website code. But I do not know what to do with it. Any help welcome.
> 
> ![funcaptcha_callback](https://user-images.githubusercontent.com/50266741/97261350-f006e000-181e-11eb-8dcc-b2f406b63dc2.png)

Execute it passing in funcap token.

**Comment by roborock** [comment]
anyone found solution to  resend token to arkose?

**Comment by Jul187** [comment]
I have the same problem. Any updates on your side? Have you been able to solve this issue ? @MacMarde

**Comment by roborock** [comment]
unfortunately not... ı havent have any new. about this.

________________________________
Gönderen: Jul187 <notifications@github.com>
Gönderildi: 14 Ocak 2021 Perşembe 17:20
Kime: ad-m/python-anticaptcha <python-anticaptcha@noreply.github.com>
Bilgi: roborock <bbs61@hotmail.com.tr>; Comment <comment@noreply.github.com>
Konu: Re: [ad-m/python-anticaptcha] Need help enter token of Funcaptcha (#86)


I have the same problem. Any updates on your side? Have you been able to solve this issue ? @MacMarde<https://github.com/MacMarde>

**Comment by k3lu0mx4c** [comment]
I have a solution

**Comment by jon-snow1999** [comment]
> I have a solution

ok wha is it

**Comment by k3lu0mx4c** [comment]
Solve outlook captcha without using service. Price 100$

**Comment by jon-snow1999** [comment]
> I think I have found the relevant website code. But I do not know what to do with it. Any help welcome.
> 
> ![funcaptcha_callback](https://user-images.githubusercontent.com/50266741/97261350-f006e000-181e-11eb-8dcc-b2f406b63dc2.png)

Also how did u find the relevant code

**Comment by FeelsBadMan1** [comment]
> 
> 
> I do not have a solution. There is someone who sells a solution:
> https://gitter.im/python-anticaptcha/
> Just ask. I would also want to have a solution but I do not want to pay 100$ for 1 line of code.

same issue here. if somebody bought it he might want to release it for free ;D

**Comment by vvazharksi** [comment]
Did anyone find solution ?

**Comment by rockey170** [comment]
Yes guys I got ,  but the problem is captcha coming on loop after completed
Let's make a telegram group and all will discuss and find a solution
Join here - https://t.me/FuncaptchaBypass
( @funcaptchabypass)
